src/js/brim/program.ts

```diff
@@ -44,35 +38,24 @@ export default function (p = "", pins: string[] = []) {
         .groupByKeys()
         .map((name) => log.tryField(name))
         .filter((f) => !!f)
-        .map(brim.syntax.include)
+        .map(syntax.include)
         .join(" ")
 
       if (/^\s*\*\s*$/.test(filter)) filter = ""
       if (newFilters.includes(filter)) filter = ""
 
-      p = stdlib.string(filter).append(" ").append(newFilters).trim().self()
-
+      p = appendWithPipe(filter, newFilters)
       return this
     },
 
     countBy(field: zed.Field) {
-      p = stdlib
-        .string(p)
-        .append(" | " + brim.syntax.countBy(field))
-        .trim()
-        .self()
-
+      p = appendWithPipe(p, syntax.countBy(field))
       return this
     },
 
     sortBy(name: string | string[], direction: "asc" | "desc" = "asc") {
-      p = stdlib
-        .string(p)
-        .replace(/\|\s*sort[^|]*$/i, "")
-        .append(" | " + brim.syntax.sortBy(name, direction))
-        .trim()
-        .self()
-
+      p = p.replace(/\|\s*sort[^|]*$/i, "")
```

Comment:
can't wait for a more surgical replace() based on indices given via the ast!

Reply:
NO DOUBT!